packages/rich-text/src/to-dom.js

```diff
@@ -132,6 +133,16 @@ function padEmptyLines( { element, createLinePadding, multilineWrapperTags } ) {
 	}
 }
 
+function prepareFormats( value ) {
+	return getFormatTypes().reduce( ( accumlator, { prepareEditableTree } ) => {
+		if ( prepareEditableTree ) {
+			return prepareEditableTree( accumlator, value.text );
```

Comment:
Any reason not to pass the entire value?

Reply:
Yes. We don't want to alter the text at this stage. Otherwise the internal value is wrong. This should be purely for applying formatting on top of text.

Reply:
We could also pass the entire value and only use the formats from what is returned. Also maybe warn if the length is different.